diag_manager/diag_manager.F90

```diff
@@ -1716,11 +1718,15 @@ LOGICAL FUNCTION diag_send_data(diag_field_id, field, time, is_in, js_in, ks_in,
        IF ( fms_error_handler('diag_manager_mod::send_data_3d', err_msg_local, err_msg) ) RETURN
     END IF
     if (use_modern_diag) then !> Set up array lengths for remapping
-      field_modern => null()
+      field_remap => null()
+      mask_remap  => null()
+      rmask_remap => null()
       ie = SIZE(field,1)
       je = SIZE(field,2)
       ke = SIZE(field,3)
-      field_modern(1:ie,1:je,1:ke,1:1) => field
+      field_remap(1:ie,1:je,1:ke,1:1) => field
+      if (present(mask)) mask_remap(1:ie,1:je,1:ke,1:1) => mask
+      if (present(rmask)) rmask_remap(1:ie,1:je,1:ke,1:1) => rmask
```

Comment:
Why isn't rmask declared with `contiguous`?

Reply:
I am not sure why the compiler didn't complain, but i declared it as`contiguous`